### PR TITLE
How to Receive Funds Using Payable Functions

### PR DESCRIPTION
**Pull request by andreipope**
How to Receive Funds Using Payable Functions

### CODE REVIEW

File: docs/en/how-to-receive-funds-using-payable-functions.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+git version 2.18.2
+```
+
+If Git is not installed, you can refer to the [Getting Started - Installing Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) page for details about installing Git.
+
+* A Rinkeby account with some ETH in it.
+
+## 1. Setting Things Up
+
+To showcase how `payable` functions, we've written a simple Solidity contract called `PayableDemo`, and added a user interface on top of it.
```
Comment (enlight) [inline docs/en/how-to-receive-funds-using-payable-functions.md]:
```suggestion
To showcase how `payable` functions work, we've written a simple Solidity contract called `PayableDemo`, and added a user interface on top of it.
```

File: docs/en/how-to-receive-funds-using-payable-functions.md [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+### 5.1 The `PayableDemo` Smart Contract
+
+The `PayableDemo` smart contract implements two functions- `buySomething` and `withdrawFunds`
+
+Let's take a look at the `buySomething` function:
+
+
+```JavaScript
+function buySomething() external payable {
+  require(msg.value == 0.001 ether, "You must send at least 0.001 ETH");
```
Comment (enlight) [inline docs/en/how-to-receive-funds-using-payable-functions.md]: The message doesn't match the condition

File: docs/en/how-to-receive-funds-using-payable-functions.md [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+* The amount of ETH sent is available in the `msg.value` variable.
+* We intentionally omitted the logic of the `buySomething` function because it's outside the scope of this tutorial. We've left a comment in the code to indicate where you should place your logic.
+
+The purpose of the `withdrawFunds` function is to let the owner withdraw funds from the contract:
+
+```JavaScript
+function withdrawFunds() public onlyOwner {
+  uint balance = address(this).balance;
+  require(balance > 0, "Balance should be > 0.");
+  address payable ownerPayable = address(uint160(owner()));
```
Comment (enlight) [inline docs/en/how-to-receive-funds-using-payable-functions.md]: This cast can be avoided by simply using `msg.sender.transfer`, `msg.sender` is supposed to be an `address payable` already

File: docs/en/how-to-receive-funds-using-payable-functions.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    process.exit(0)
+  }
+  filterEvents(payableDemoContract, web3js)
+  try {
+    await payableDemoContract.methods.withdrawFunds().send({ from: account })
+  } catch (error) {
+    console.log('Something went wrong.')
+  }
+  const ownerBalanceInWei = await ethCoin.getBalanceOfAsync(Address.fromString(`${client.chainId}:${account}`))
+  console.log('Owner\'s balance: ' + web3js.utils.fromWei(ownerBalanceInWei.toString(), 'ether'))
```
Comment (enlight) [inline docs/en/how-to-receive-funds-using-payable-functions.md]:
```suggestion
  console.log(`Owner's balance: ${web3js.utils.fromWei(ownerBalanceInWei.toString(), 'ether')}`)
```

File: docs/en/how-to-receive-funds-using-payable-functions.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+  if (payableDemoContractBalance.toString() === '0') {
+    console.log('The balance of the contract is 0. Nothing to withdraw.')
+    client.disconnect()
+    process.exit(0)
+  }
+  filterEvents(payableDemoContract, web3js)
+  try {
+    await payableDemoContract.methods.withdrawFunds().send({ from: account })
+  } catch (error) {
+    console.log('Something went wrong.')
```
Comment (enlight) [inline docs/en/how-to-receive-funds-using-payable-functions.md]: Should log the actual error too.

File: docs/en/how-to-receive-funds-using-payable-functions.md (hunk 3 above)
Comment (andreipope) [reply docs/en/how-to-receive-funds-using-payable-functions.md]: @enlight Right! Thanks 👍

File: docs/en/how-to-receive-funds-using-payable-functions.md (hunk 5 above)
Comment (andreipope) [reply docs/en/how-to-receive-funds-using-payable-functions.md]: Nice catch. Thanks!

File: docs/en/how-to-receive-funds-using-payable-functions.md (hunk 2 above)
Comment (andreipope) [reply docs/en/how-to-receive-funds-using-payable-functions.md]: Fixed.

**Comment by andreipope** [comment]
@enlight It's updated.

**Pull request merged**